server/src/main/java/org/apache/iotdb/db/mpp/sql/planner/plan/node/metedata/read/DevicesSchemaScanNode.java

```diff
@@ -66,22 +62,8 @@ public PlanNode clone() {
   }
 
   @Override
-  public int allowedChildCount() {
-    return NO_CHILD_ALLOWED;
-  }
-
-  @Override
-  public List<String> getOutputColumnNames() {
-    if (hasSgCol) {
-      return Arrays.asList(COLUMN_DEVICES, COLUMN_STORAGE_GROUP, COLUMN_IS_ALIGNED);
-    }
-    return Arrays.asList(COLUMN_DEVICES, COLUMN_IS_ALIGNED);
-  }
-
-  @Override
-  public void serialize(ByteBuffer byteBuffer) {
+  protected void serializeAttributes(ByteBuffer byteBuffer) {
     PlanNodeType.DEVICES_SCHEMA_SCAN.serialize(byteBuffer);
-    ReadWriteIOUtils.write(getPlanNodeId().getId(), byteBuffer);
     ReadWriteIOUtils.write(path.getFullPath(), byteBuffer);
     ReadWriteIOUtils.write(limit, byteBuffer);
     ReadWriteIOUtils.write(offset, byteBuffer);
```

Comment:
1. If the parent class maintains the serialization work of its own attributes, then it should also take over the deserialization work so that sub-classes need no care about the ser and deser order of attributes in parent class.
If so, when doing deserialization, the parent should return an instance contains all deserialized attributes, but the parent is an `abstract` class, we cannot just simply use `new xxx`.
2. If some sub-classes do not want to serialize all attributes in parent, it will always need to implements its own serialize function as needed.

Reply:
fine